crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -281,6 +281,42 @@ def prepare(options = {})
 
       protected
 
+      def upgrade_first_cluster_node(node_name, next_node_name)
```

Comment:
I believe ```next_node_name``` is a bit of a misnomer here. Technically it doesn't really matter whether the node passed there is upgraded next after the first node (I guess the only thing you need here is a cluster member that still running pacemaker?). Additionally the implementation also doesn't really make sure it's upgraded next (AFAICS, ```upgrade_cluster_nodes``` simply reruns the search for non-upgrade nodes, and I believe chef doesn't make any assumptions in which order they will be returned).